### PR TITLE
keep topology defn during update

### PR DESCRIPTION
**Pull request by huijunw**
We found that the topology defn is changed during update process, which causes the messy state. The original topology definition should remain same during the whole topology lifecycle, while the packing plan shows the latest topology information during update/repack.

This PR keeps topology defn during update.

Tested in local scheduler with #2600.

### CODE REVIEW

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java [hunk 1]
```diff
@@ -163,10 +163,6 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       scalableScheduler.get().addContainers(containerDelta.getContainersToAdd());
     }
 
-    // update parallelism in updatedTopology since TMaster checks that
-    // Sum(parallelism) == Sum(instances)
-    logInfo("Update new Topology: %s", stateManager.updateTopology(updatedTopology, topologyName));
```
Comment (maosongfu) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: If we remove the topology-update on stateManager, it seems we could remove or clean a lot of codes, like:
```
private TopologyAPI.Topology getUpdatedTopology(...)
static TopologyAPI.Topology mergeTopology(...)
private static boolean updateComponent(...)
```
And a lot of unit tests.
Can we also do that?

Comment (huijunw) [reply heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: removed

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java [hunk 2]
```diff
@@ -147,14 +145,10 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
...
 
-    // fetch the topology, which will need to be updated
-    TopologyAPI.Topology updatedTopology =
-        getUpdatedTopology(topologyName, proposedPackingPlan, stateManager);
-
     // deactivate and sleep
     if (initiallyRunning) {
       // Update the topology since the state should have changed from RUNNING to PAUSED
-      updatedTopology = deactivateTopology(stateManager, updatedTopology, proposedPackingPlan);
+      deactivateTopology(stateManager, topology, proposedPackingPlan);
```
Comment (maosongfu) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: Good to have a comment here saying:
`Will throw exceptions internally if failed to xxxx`

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java [hunk 3]
```diff
@@ -177,7 +167,7 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       // delete the packing plan. Instead we could message tmaster to invalidate the physical plan
       // and/or possibly even update the packing plan directly
       SysUtils.sleep(Duration.ofSeconds(10));
-      reactivateTopology(stateManager, updatedTopology, removableContainerCount);
+      reactivateTopology(stateManager, topology, removableContainerCount);
```
Comment (maosongfu) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: Good to have a comment here saying:
`Will throw exceptions internally if failed to xxxx`

**Comment by maosongfu** [comment]
Also, plz make sure no unit tests will break

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java (hunk 2 above)
Comment (huijunw) [reply heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: done

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java (hunk 3 above)
Comment (huijunw) [reply heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java]: done

**Pull request merged**